xc/xc7/tests/dram/CMakeLists.txt

```
@@ -1,24 +1,20 @@
 add_custom_target(test_dram_packing)
 
-function(dram_tests dram_num_instances dram_mode outpad inpad enable_vivado_target)
+function(dram_tests dram_num_instances dram_mode outpad inpad blk_tl_slicem)
```

You added the documentation `# dram_tests(<dram_num_instances> <dram_mode> <syn_outpad_assert_usage> <syn_inpad_assert_usage> <blk_tl_slicem_assert_usage>)` below which seems to indicate to me that these arguments would be better named something like;
```suggestion
function(dram_tests dram_num_instances dram_mode syn_outpad_assert_usage syn_inpad_assert_usage blk_tl_slicem_assert_usage)
```

Some other options,
```suggestion
function(dram_tests dram_num_instances dram_mode outpad_num_tiles inpad_num_tiles slicem_num_tiles)
```

or maybe

```suggestion
function(dram_tests dram_num_instances dram_mode assert_outpad_num_tiles assert_inpad_num_tiles assert_slicem_num_tiles)
```

---

Or maybe we should let people just specify the `ASSERT_BLOCK_TYPES_ARE_USED` directly so that;
```
dram_tests(4 32x1s 17 18 1 Y)
```
becomes
```
dram_tests(4 32x1s SYN-OUTPAD=17,SYN-INPAD=18,BLK-TL-SLICEM=1)
```

---

I really like the last solution. It gives more flexibility and it is easier to read. I applied this change and included nicer documentation of `DRAM_TESTS` function.